screeps-game-api/src/raw_memory.rs

```diff
@@ -26,13 +23,47 @@ pub fn set_active_segments(ids: &[i32]) {
     }
 }
 
-pub fn set_active_foreign_segment(username: &str, id: Option<i32>) {
-    match id {
-        Some(id) => js! { RawMemory.setActiveForeignSegment(@{username}, @{id}); },
-        None => js! { RawMemory.setActiveForeignSegment(@{username}); },
+get_from_js!(get_segment(id: u32) -> { 
+    RawMemory.segments[@{id}] 
+} -> Option<String>);
+
+pub fn set_segment(id: i32, data: &str) {
+    js! {
+        RawMemory.segments[@{id}] = @{data};
+    }
+}
+
+// pub fn get_foreign_segment() -> ForeignSegment {
+//     let v = js! {
+//         return RawMemory.foreignSegment;
+//     };
+    
+//     v.try_into().unwrap()
+// }
```

Comment:
If you're alright with it I think this'd be better just removed rather than commented out. It can stay in git history if we need it.

Reply:
Good catch, this wasn't meant to stay there.